src/pymor/algorithms/rand_la.py

```diff
@@ -0,0 +1,362 @@
+        If 'False' the methods requires two sets of matvecs Ax.
+    sigma
+        If not `None` transforms the eigenvalue problem such that the k eigenvalues
+        closest to sigma are computed.
+    which
+        A string specifying which `k` eigenvalues and eigenvectors to compute:
+
+        - `'LM'`: select eigenvalues with largest magnitude
+        - `'SM'`: select eigenvalues with smallest magnitude
+        - `'LR'`: select eigenvalues with largest real part
+        - `'SR'`: select eigenvalues with smallest real part
+        - `'LI'`: select eigenvalues with largest imaginary part
+        - `'SI'`: select eigenvalues with smallest imaginary part
+    b
+        Initial vector for Arnoldi iteration. Default is a random vector.
+    l
+        The size of the Arnoldi factorization. Default is `min(n - 1, max(2*k + 1, 20))`.
+    maxiter
+        The maximum number of iterations.
+    complex_evp
+        Wether to consider an eigenvalue problem with complex operators. When operators
+        are real setting this argument to `False` will increase stability and performance.
+    left_evp
+        If set to `True` compute left eigenvectors else compute right eigenvectors.
+    seed
+        Random seed which is used for computing the initial vector for the Arnoldi
+        iteration.
+
+    Returns
+    -------
+    w
+        A 1D |NumPy array| which contains the computed eigenvalues.
+    V
+        A |VectorArray| which contains the computed eigenvectors.
+    """
+    logger = getLogger('pymor.algorithms.rand_la')
+
+    assert isinstance(A, Operator) and A.linear
+    assert not A.parametric
+    assert A.source == A.range
+    assert 0 <= modes <= max(A.source.dim, A.range.dim) and isinstance(modes, int)
+    assert p >= 0 and isinstance(p, int)
+    assert q >= 0 and isinstance(q, int)
+    
+    if E is None:
+        E = IdentityOperator(A.source)
+    else:
+        assert isinstance(E, Operator) and E.linear
+        assert not E.parametric
+        assert E.source == E.range
+        assert E.source == A.source
+    
+    if A.source.dim == 0 or A.range.dim == 0:
+        return A.source.empty(), np.array([]), A.range.empty()
+    
+    C = InverseOperator(E) @ A
+    Y, Omega = rrf(C, q=q, l=modes+p, return_rand=True)
+    Omega_op = VectorArrayOperator(Omega)
+    Q = gram_schmidt(Y, product=E)
+    Q_op = VectorArrayOperator(Q)
+    
+    if single_pass:
+        X = VectorArrayOperator(Omega_op.apply_adjoint(E.apply(Q)))
+        Z = VectorArrayOperator(Q_op.apply_adjoint(E.apply(Omega)))
+        T = InverseOperator(X) @ VectorArrayOperator(Omega_op.apply_adjoint(A.apply(Q))) @ InverseOperator(Z)
+    else: 
+        T = Q_op.H @ A @ Q_op
+        
+    w, S = eigs(T, E=None, k=modes, sigma=sigma, which=which, b=b, l=l, maxiter=maxiter, tol=tol,
+         imag_tol=imag_tol, complex_pair_tol=complex_pair_tol, complex_evp=complex_evp, left_evp=left_evp, seed=seed)
```

Comment:
The number of arguments here and in the function header is very long. Maybe its worth to consider introducing a `eigs_solver_options` variable or something similar.

Reply:
Does it actually make sense to call an iterative solver here? `T` is a `(modes+p)x(modes+p)` matrix. Also, it should be hermitian, if I'm not mistaken, so `eigh` should be the method of choice?